cg/models/flow_cell/flow_cell.py

```diff
@@ -101,10 +101,35 @@ def get_sample_sheet_path_hk(self) -> Path:
         return self._sample_sheet_path_hk
 
     def get_flow_cell_run_dir(self) -> Path:
-        """Return the flow cells run directory if the FlowCellsDirectoryData was initialised with Demultiplexed runs dir."""
+        """
+        Return the flow cells run directory
+        if the FlowCellsDirectoryData was initialised with Demultiplexed runs dir.
+        """
         current_path: str = self.path.as_posix()
-        if DEMULTIPLEXED_RUNS in current_path:
-            return Path(str.replace(current_path, DEMULTIPLEXED_RUNS, FLOW_CELL_RUNS))
+        if DemultiplexingDirsAndFiles.DEMULTIPLEXED_RUNS_DIRECTORY_NAME in current_path:
+            return Path(
+                str.replace(
+                    current_path,
+                    DemultiplexingDirsAndFiles.DEMULTIPLEXED_RUNS_DIRECTORY_NAME,
+                    DemultiplexingDirsAndFiles.FLOW_CELLS_DIRECTORY_NAME,
+                )
+            )
+        return self.path
+
+    def get_demultiplexed_runs_dir(self) -> Path:
+        """
+        Return the flow cells run directory
+        if the FlowCellsDirectoryData was initialised with Demultiplexed runs dir.
```

Comment:
same here, is much clearer
```suggestion
        Return the demultiplexed run directory regardless of the path used to initialise the FlowCellsDirectoryData.
```